exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/logs/OtlpStdoutLogRecordExporterBuilder.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.logging.otlp.internal.logs;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.exporter.logging.otlp.OtlpJsonLoggingLogRecordExporter;
+import io.opentelemetry.exporter.logging.otlp.internal.writer.JsonWriter;
+import io.opentelemetry.exporter.logging.otlp.internal.writer.LoggerJsonWriter;
+import io.opentelemetry.exporter.logging.otlp.internal.writer.StreamJsonWriter;
+import java.io.OutputStream;
+import java.util.logging.Logger;
+
+/**
+ * Builder for {@link OtlpJsonLoggingLogRecordExporter}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class OtlpStdoutLogRecordExporterBuilder {
+
+  private static final String TYPE = "log records";
+
+  private Logger logger;
+  private JsonWriter jsonWriter;
+  private boolean wrapperJsonObject = true;
+
+  public OtlpStdoutLogRecordExporterBuilder(Logger logger) {
+    this.logger = logger;
+    this.jsonWriter = new LoggerJsonWriter(logger, TYPE);
+  }
+
+  /**
+   * Sets the exporter to use the specified JSON object wrapper.
+   *
+   * @param wrapperJsonObject whether to wrap the JSON object in an outer JSON "resourceLogs"
+   *     object.
+   */
+  public OtlpStdoutLogRecordExporterBuilder setWrapperJsonObject(boolean wrapperJsonObject) {
+    this.wrapperJsonObject = wrapperJsonObject;
+    return this;
+  }
+
+  /**
+   * Sets the exporter to use the specified output stream.
+   *
+   * @param outputStream the output stream to use.
+   */
+  public OtlpStdoutLogRecordExporterBuilder setOutputStream(OutputStream outputStream) {
+    requireNonNull(outputStream, "outputStream");
+    this.jsonWriter = new StreamJsonWriter(outputStream, TYPE);
+    return this;
+  }
+
+  /** Sets the exporter to use the specified logger. */
+  public OtlpStdoutLogRecordExporterBuilder setLogger(Logger logger) {
+    requireNonNull(logger, "logger");
+    this.logger = logger;
+    this.jsonWriter = new LoggerJsonWriter(logger, TYPE);
```

Comment:
WDYT about removing this? Logger is serving double duty right now: we log error / warnings to it, and we log the output JSON to it. But we can break this overload by requiring users to be explicit:

* Call `setLogger(..)` to configure the logger used for logging errors / warnings. This is really an implementation detail which users shouldn't ever call.
* Call `setOutputStream(..)` to configure where the output JSON is written to. 
* If by chance you want to write the output json and errors / warnings to the same logger, call both.

```suggestion
```

Reply:
we can remove the entire method if we accept that `OtlpStdoutLogRecordExporter` can never write OTLP to a logger - which I actually had in mind in the beginning.

I also would suggest to remove `setWrapperJsonObject` from the builder (or make it private) - I don't think it's needed.

Reply:
> we can remove the entire method if we accept that OtlpStdoutLogRecordExporter can never write OTLP to a logger - which I actually had in mind in the beginning.

Is this because there's no way to call `setOutputstream(..)` with an OutputStream which writes to a logger?

Reply:
yes

Reply:
Another variation of my suggestion is to have methods:
- `setLogger(Logger)` - specify the logger used for logging errors / warnings
- `setOutput(Logger)` - specify that the json should be output to the logger
- `setOutput(OutputStream)` - specify that the json should be output to the OutputStream

This clearly separates the logger used for errors / warnings from the one used for outputting json. And it makes it clear that `setOutput(Logger)` and `setOutput(OutputStream)` are overloads - different mechanisms for specifying where the json should be output too.

Reply:
I like that :smile: 

do we even need `setLogger` in this case?